### DISCUSSION TITLE
Supports collect Nebula Graph cluster metrics through NGQL statements

### DISCUSSION THREAD

**Discussion by LiuTianyou** [apache/hertzbeat#1866] [Ideas]
The NebulaGraph monitoring currently supported by HertzBeat collects performance indicator data through the node's open monitoring HTTP API, which is usually for a single node and does not include cluster information.We can support collecting more data through NGQL statement query.

Metrics that can be queried include but are not limited to:

##### Cluster level:
- `Charset` and `Collation` used by the cluster
- Number of meta nodes (ONLINE/OFFLINE) in the cluster
- Number of graph nodes (ONLINE/OFFLINE) in the cluster
- Number of storage nodes (ONLINE/OFFLINE) in the cluster
- The number of sessions connected to the cluster
- Number of queries being executed
- Number of background jobs being executed
- Number of graph spaces in the cluster
- Meta cluster leader information
- ...

##### Graph space level:
- Basic information of graph space
- label, number of nodes
- The number of background jobs running in the current graph space
- The number of indexes in the current graph space
- The number of jobs being executed in the current graph space
- ...

#### Custom monitoring template
Users can use the NGQL protocol in custom templates to connect to the NebulaGraph cluster and query the indicators they care about.

#### About  licenses
Nebula java client uses the Apache 2.0 licenses

**Comment by tomsun28** [comment]
This looks very good, we can define a new protocol for Nebula Graph to implement it. 👍👍 
As you say, the new protocol can query any data from NebulaGraph-compatible system.
